File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/document/mongo/Token.java
package com.raxim.myscoutee.profile.data.document.mongo;

import java.util.UUID;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

@Document(collection = "tokens")
public class Token {
    @Id
    @JsonIgnore
    private UUID id;

    @JsonProperty(value = "deviceKey")
    private String deviceKey;

    // mqtt, firebase
    @JsonIgnore
    private String type;

    //userUUId
    @JsonIgnore
    private UUID uuid;

    public UUID getUuid() {
        return uuid;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    @JsonIgnore
    public String getDeviceKey() {
        return deviceKey;
    }

    public void setDeviceKey(String deviceKey) {
        this.deviceKey = deviceKey;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public UUID getId() {
        return id;
    }
}

File: server/projects/profile/src/main/java/com/raxim/myscoutee/profile/data/document/mongo/Token.kt
package com.raxim.myscoutee.profile.data.document.mongo

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import com.querydsl.core.annotations.QueryEntity
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.*

// index on uuid

@QueryEntity
@Document(collection = "tokens")
data class Token(
    @Id
    @field:JsonIgnore
    val id: UUID? = UUID.randomUUID(),

    // device registration key
    @get:JsonIgnore
    @set:JsonProperty(value = "deviceKey")
    var deviceKey: String? = null,

    //user uuid
    @field:JsonIgnore
    val uuid: UUID? = null
)